File: VB/RegistrationForm.Lesson3/ViewModel/RecordsViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.DataAnnotations
Imports DevExpress.Mvvm.POCO
Imports RegistrationForm.DataModel
Imports System.Collections.Generic

Namespace RegistrationForm.ViewModel
    <POCOViewModel> _
    Public Class RecordsViewModel
        Public Shared Function Create() As RecordsViewModel
            Return ViewModelSource.Create(Function() New RecordsViewModel())
        End Function
        Protected Sub New()
            Messenger.Default.Register(Of DBEmployeesChangedMessage)(Me, AddressOf OnDBEmployeesChanged)
            Employees = New List(Of Employee)()
            If Not Me.IsInDesignMode() Then
                InitializeEmployees()
            End If
        End Sub
        Private Sub InitializeEmployees()
            Employees = EmployeesModelHelper.GetEmployees()
        End Sub
        Private Sub OnDBEmployeesChanged(ByVal message As DBEmployeesChangedMessage)
            InitializeEmployees()
        End Sub

        Public Property Employees() As List(Of Employee)
    End Class
End Namespace

File: CS/RegistrationForm.Lesson5/ViewModel/RecordsViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.DataAnnotations;
using DevExpress.Mvvm.POCO;
using RegistrationForm.DataModel;
using System.Collections.Generic;

namespace RegistrationForm.ViewModel {
    [POCOViewModel]
    public class RecordsViewModel {
        public static RecordsViewModel Create() {
            return ViewModelSource.Create(() => new RecordsViewModel());
        }
        protected RecordsViewModel() {
            Messenger.Default.Register<DBEmployeesChangedMessage>(this, OnDBEmployeesChanged);
            Employees = new List<Employee>();
            if(!this.IsInDesignMode())
                InitializeEmployees();
        }
        void InitializeEmployees() {
            Employees = EmployeesModelHelper.GetEmployees();
        }
        void OnDBEmployeesChanged(DBEmployeesChangedMessage message) {
            InitializeEmployees();
        }

        public virtual List<Employee> Employees { get; set; }
    }
}